src/base/bittorrent/session.cpp

```diff
@@ -3677,11 +3697,8 @@ void Session::handleTorrentChecked(TorrentHandle *const torrent)
 
 void Session::handleTorrentFinished(TorrentHandle *const torrent)
 {
-    if (!torrent->hasError() && !torrent->hasMissingFiles()) {
+    if (!torrent->hasError() && !torrent->hasMissingFiles())
         saveTorrentResumeData(torrent);
-        if (isQueueingSystemEnabled())
-            handleTorrentsPrioritiesChanged();
```

Comment:
Shouldn't you add `saveTorrentsQueue()` here too?

Reply:
It's actually very difficult to keep track of all the events when the queue can be changed, so I save it when the session ends. In the event of a crash, I also save the queue in cases where it may affect the correct loading.
There is no critical point here so I prefer skip additional disk IO.

Reply:
OK